crates/networking/p2p/rlpx/eth/transactions.rs

```diff
@@ -182,20 +239,37 @@ impl RLPxMessage for GetPooledTransactions {
 }
 
 // https://github.com/ethereum/devp2p/blob/master/caps/eth.md#pooledtransactions-0x0a
+#[derive(Debug)]
 pub(crate) struct PooledTransactions {
     // id is a u64 chosen by the requesting peer, the responding peer must mirror the value for the response
     // https://github.com/ethereum/devp2p/blob/master/caps/eth.md#protocol-messages
     id: u64,
-    pooled_transactions: Vec<Transaction>,
+    pooled_transactions: Vec<P2PTransaction>,
 }
 
 impl PooledTransactions {
-    pub fn new(id: u64, pooled_transactions: Vec<Transaction>) -> Self {
+    pub fn new(id: u64, pooled_transactions: Vec<P2PTransaction>) -> Self {
         Self {
             pooled_transactions,
             id,
         }
     }
+
+    /// Saves every incoming pooled transaction to the mempool.
+    pub fn handle(&self, store: &Store) -> Result<(), MempoolError> {
```

Comment:
I think here we can, instead, take ownership of self (or the inner vec) to avoid cloning each transaction. It's fine like this, and taking ownership could result in opening an awful can of worms, but it's worth the try.